File: python_algorithms/utility/communication.py
from flask import request
from utility.algorithmExecutor import init_execution

class Datos:
    def __init__(self, testID, matrixA, matrixB):
        self.TestID = testID
        self.MatrixA = matrixA
        self.MatrixB = matrixB

def Init_PythonExecution(request):
    data = request.get_json()

    if not data:
        return "Error al leer el cuerpo de la solicitud", 400

    try:
        datos = Datos(data['TestID'], data['MatrixA'], data['MatrixB'])
    except KeyError:
        return "Error al decodificar los datos JSON", 400

    test = datos.TestID
    matrizA = datos.MatrixA
    matrizB = datos.MatrixB

    # Llama a la función init_execution con las matrices y el test ID
    init_execution(matrizA, matrizB, test)

    return "OK - Correctly recorded execution times", 200

File: go_algorithms/utility/communication.go
package utility

import (
	"encoding/json"
	"io/ioutil"
	"net/http"
)

type Datos struct {
	TestID  string  `json:"TestID"`
	MatrixA [][]int `json:"MatrixA"`
	MatrixB [][]int `json:"MatrixB"`
}

func Init_GolangExecution(w http.ResponseWriter, r *http.Request) {

	body, err := ioutil.ReadAll(r.Body)
	if err != nil {
		http.Error(w, "Error al leer el cuerpo de la solicitud", http.StatusBadRequest)
		return
	}

	var datos Datos
	if err := json.Unmarshal(body, &datos); err != nil {
		http.Error(w, "Error al decodificar los datos JSON", http.StatusBadRequest)
		return
	}

	test := datos.TestID
	matrizA := datos.MatrixA
	matrizB := datos.MatrixB

	Init_execution(matrizA, matrizB, test)

	w.WriteHeader(http.StatusOK)
	w.Write([]byte("OK - Correctly recorded execution times"))
}
